woocommerce/woocommerce-gateway-stripe
# Use deferred intents on the Pay for Order page

Fixes #2766 

## Changes proposed in this Pull Request:

This PR utilises the changes introduced in `add/my-account-add-card-with-deferred-intent` to add the deferred intent payment elements to the Pay for Order page. 

These payment elements are the same ones used on the classic checkout. 

## Testing instructions

1. Go to **WooCommerce → Orders → Add new**
2. Add items, a customer and save. Leave the order as pending payment.
3. Click the **Customer payment page →** link.
4. On your Stripe dashboard go to the payments table and note that a payment intent has been created and is in-complete.

<img width="500" alt="Screenshot 2023-12-15 at 10 06 00 am" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/4829b556-89fe-41d4-afc7-77ebfa9b9969">

5. Checkout this branch (`issue/2766-deferred-intent-pay-for-order-m2`) and re-`build`
6. Repeat from step 3. 
   - Optional, add more items or change the order so the total is different so the two intents are distinguishable. 
7. This time, when you view the Stripe Dashboard, there won't be a new payment intent.
8. Go back to the Pay for Order page and complete the checkout.
9. Note that order is paid and the payment was taken in Stripe.

<img width="500" alt="Screenshot 2023-12-15 at 10 22 37 am" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/40e0817e-e4ff-4b5c-824b-75ea003f6ac5">

<img width="400" alt="Screenshot 2023-12-15 at 10 22 54 am" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/3ff45986-2650-42af-ae74-449442a89f48">

### Other tests

**SCA/3DS**

- Will be fixed with https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2797.

**Alternative payment methods**

Because these payment elements use the same ones implemented in the original classic checkout PR, it's still a unified payment element and all alternative methods appear like this:

<img width="600" alt="Screenshot 2023-12-15 at 1 10 49 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/04648037-8fed-47ba-9233-f455911228ce">

**Auth and capture**

1. Go to **WooCommerce → Settings → Payments → Stripe → Settings**
2. Enable Auth and capture. 
3. Repeat the steps above to pay for an order via the Pay for Order page.
4. View the admin edit order screen and note the order is on-hold and has an order note about auth and capture.
5. Capture the payment by changing the order status to processing or completed. 
6. Verify the payment was taken authorised and captured in your Stripe dashboard. 

<img width="276" alt="Screenshot 2023-12-15 at 1 14 07 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/387145cf-58c6-4625-ba47-c175df370f2a">

---


**Post merge**
